views/default/js/elgg/admin/upgrade.js

```diff
@@ -0,0 +1,225 @@
+
+define(function(require) {
+	var $ = require('jquery');
+	var elgg = require('elgg');
+
+	var upgrades = $('.elgg-upgrade');
+	var upgrade;
+	var className;
+	var progressbar;
+	var upgradeStartTime;
+	var timer;
+	var counter;
+	var percent;
+	var errorMessages = [];
+	var offset = 0;
+	var numSuccess;
+	var numError;
+	var numProcessed = 0;
+	var total;
+
+	/**
+	 *
+	 */
+	init = function() {
+		$('#upgrade-run').click(run);
+
+		upgrades.each(function(key, value) {
+			// Initialize progressbar
+			$(value).find('.elgg-progressbar').progressbar();
+		});
+	};
+
+	/**
+	 * Runs all individual upgrades one at a time
+	 *
+	 * @param {Object} e Event object
+	 */
+	var run = function(e) {
+		e.preventDefault();
+
+		// Replace button with spinner when upgrade starts
+		$('#upgrade-run').addClass('hidden');
+		$('#upgrade-spinner').removeClass('hidden');
```

Comment:
Oh, right. It's displayed with fixed location on top of the screen, right? So no need to include it in the actual elements in the upgrade list layout.